tests/components/RangeInput.test.tsx

```diff
@@ -0,0 +1,296 @@
+      displayName: 'Up to 20',
+      fieldId: '123',
+      matcher: '$between',
+      selected: true,
+      value: {
+        end: {
+          matcher: '$le',
+          value: 20
+        },
+      }
+    });
+  });
+
+  it('renders correctly when clearing input', async () => {
+    render(
+      <FilterGroupContext.Provider value={filterGroupContextValue}>
+        <FiltersContext.Provider value={filterContextValue}>
+          <RangeInput />
+        </FiltersContext.Provider>
+      </FilterGroupContext.Provider>);
+    const maxTextbox = screen.getAllByRole('textbox')[1];
+    userEvent.type(maxTextbox, '20');
+    await waitFor(() => {
+      expect(maxTextbox).toHaveValue('20');
+    });
+    expect(screen.getByText('Clear min and max')).toBeDefined();
+    userEvent.click(screen.getByText('Clear min and max'));
+    expect(maxTextbox).toHaveValue('');
+  });
+});
+
+describe('Renders correctly for min and max inputs', () => {
+  beforeEach(() => {
+    mockAnswersHooks({ mockedState, mockedActions });
+  });
+  it('renders input value, clear, and apply option when inputing min and max values', async () => {
+    render(
+      <FilterGroupContext.Provider value={filterGroupContextValue}>
+        <FiltersContext.Provider value={filterContextValue}>
+          <RangeInput />
+        </FiltersContext.Provider>
+      </FilterGroupContext.Provider>);
+    const [minTextbox, maxTextbox] = screen.getAllByRole('textbox');
+    userEvent.type(minTextbox, '10');
+    userEvent.type(maxTextbox, '20');
+    await waitFor(() => {
+      expect(minTextbox).toHaveValue('10');
+    });
+    expect(maxTextbox).toHaveValue('20');
+    expect(screen.getByText('Clear min and max')).toBeDefined();
+    expect(screen.getByText('Apply')).toBeDefined();
+  });
+
+  it('record proper values in state when applying min and max', async () => {
+    render(
+      <FilterGroupContext.Provider value={filterGroupContextValue}>
+        <FiltersContext.Provider value={filterContextValue}>
+          <RangeInput />
+        </FiltersContext.Provider>
+      </FilterGroupContext.Provider>);
+    const actions = spyOnActions();
+    const [minTextbox, maxTextbox] = screen.getAllByRole('textbox');
+    userEvent.type(minTextbox, '10');
+    userEvent.type(maxTextbox, '20');
+    await waitFor(() => {
+      expect(minTextbox).toHaveValue('10');
+    });
+    expect(maxTextbox).toHaveValue('20');
+    expect(screen.getByText('Apply')).toBeDefined();
+    userEvent.click(screen.getByText('Apply'));
+    expect(actions.setFilterOption).toHaveBeenCalledWith({
+      displayName: '10 - 20',
+      fieldId: '123',
+      matcher: '$between',
+      selected: true,
+      value: {
+        start: {
+          matcher: '$ge',
+          value: 10
+        },
+        end: {
+          matcher: '$le',
+          value: 20
+        },
+      }
+    });
+  });
+
+  it('renders correctly when clearing inputs', async () => {
```

Comment:
can we ensures that actions.setFilterOption is not called at the end? (similarly for other `renders correctly when clearing input` tests)